runtime/src/main/cpp/Memory.cpp

```diff
@@ -1476,6 +1409,8 @@ void processDecrements(MemoryState* state) {
      toRelease->pop_back();
      if (isMarkedAsRemoved(container))
        continue;
+     if (container->shareable())
+       container = realShareableContainer(container);
```

Comment:
Why not enqueue `ObjHeader*` instead?

Reply:
Not a big deal, I guess. We generally try to use containers in MM algorithms.